src/BinSkim.Rules/EnableSpectreMitigations.cs

```diff
@@ -135,196 +132,197 @@ public override void Analyze(BinaryAnalyzerContext context)
 
-                    default:
-                        // TODO-paddymcd-MSFT: Add warning for unrecognized languages
-                        // built with unrecognized compiler.
-                        badModuleList.Add(
-                            om.CreateCompilandRecordWithSuffix(
-                                String.Format(CultureInfo.InvariantCulture,
-                                RuleResources.BA2024_Error_BuiildWithSpectreMitigation_UnrecognizedCompiler)));
-                        continue;
+                    default:
+                    {
+                        // TODO: https://github.com/Microsoft/binskim/issues/117
+                        // Review unknown languages for this and all checks
+                    }
+                    continue;
                 }
 
-                // Get the appropriate compiler Version against which to check this compiland
+                // Get the appropriate compiler version against which to check this compiland.
                 bool supportsQspectre = false;
                 bool supportsd2guardspecload = false;
 
                 // check that we are greater than or equal to the first fully supported release: 15.6 first
                 Version omVer = omDetails.CompilerVersion;
-                if (omVer >= minimumCompilers[VS2017_15_6_PREV4].compilerVersion)
+                if (omVer >= minimumCompilers[VS2017_15_6_PREV4].CompilerVersion)
                 {
                     supportsQspectre = minimumCompilers[VS2017_15_6_PREV4].QSpectre;
-                    supportsd2guardspecload = minimumCompilers[VS2017_15_6_PREV4].d2specguard;
+                    supportsd2guardspecload = minimumCompilers[VS2017_15_6_PREV4].D2SpecGuard;
                 }
                 else
                 {
                     // Now check the patched versions that we match on the major, minor versions and then are greater than or equal to on the rest...
                     foreach (var compilerVersionEntry in minimumCompilers)
                     {
-                        Version ver = compilerVersionEntry.Value.compilerVersion;
+                        Version version = compilerVersionEntry.Value.CompilerVersion;
 
-                        if (ver.Major == omVer.Major
-                            && ver.Minor == omVer.Minor 
-                            && ver >= omVer)
+                        if (version.Major == omVer.Major &&
+                            version.Minor == omVer.Minor &&
+                            version >= omVer)
                         {
                             supportsQspectre = compilerVersionEntry.Value.QSpectre;
-                            supportsd2guardspecload = compilerVersionEntry.Value.d2specguard;
+                            supportsd2guardspecload = compilerVersionEntry.Value.D2SpecGuard;
                         }
                     }
                 }
 
                 if (!supportsd2guardspecload && !supportsQspectre)
```

Comment:
If we want BA2006 to fire here, we will need to update the version that it requires--I believe the check requires VS2013 right now.  We'd need to bump to 2015.3-(QSpectre patch).

Reply:
Agree--however, the design of BinSkim doesn't, at the moment, allow folks to opt in/out of rules.  Per Michael, this was deliberate--it separates the reporting layer (BinSkim) from a separate policy layer (e.x. tools that can consume the results and act on them).

@michaelcfanning I do agree with Paddy that if we aren't going to bump the version in BA2006, we should report this here.